src/output_queue/output_queue.py

```diff
@@ -86,12 +96,20 @@ def _check_priority_queue(self):
         now = time.time()
 
         try:
-            while not self.queue.empty() and now >= self.queue.peek().timestamp:
+            while now >= self.queue.peek().timestamp:
                 midiEvent = self.queue.get()
 
+                if midiEvent.event.type == 'note_off' or midiEvent.event.velocity == 0:
+                    if midiEvent.event.note in self._notePlayingSet.keys():
+                        del self._notePlayingSet[midiEvent.event.note]
+                elif midiEvent.event.type == 'note_on':
+                    self._notePlayingSet[midiEvent.event.note] = True
+
                 self._open_port.send(midiEvent.event)
         except IndexError:
-            pass # Expected when pausing/stopping and the queue is cleared
+            pass # Expected when the queue is empty
+        except AttributeError:
```

Comment:
You can sell me on the index error, but why is this expected when empty?

Reply:
This now happens because the mixing system can preempt here in between `peek()` and `get()` and clear the queue right before `get()` is called. Updated the comment to explain this.